### DISCUSSION TITLE
It is possible to use a consolonia control in an avalonia project

### DISCUSSION THREAD

**Discussion by guilhermeolisi** [Consolonia/Consolonia#101] [General]
Hello, first of all congratulations on your project, very interesting
Is it possible to use a consolonia control in a regular avalonia project? I need a console like control and I think using a consolonia text block instead of a regular avalonia text block in my project can improve performance

**Comment by jinek** [comment]
Hello, thank you!
Consolonia does not have own controls. Every control which you see in Consolonia is just plain Avalonia control with only different (simplified) template. Thus, it does not make big difference if you run TextBlock on Consolonia or Avalonia. Text rendering is somehow faster, probably, but it's only because in Consolonia we can not choose fonts, size, effects etc. Primary purpose of Consolonia is to run headless with lower resources. For example, on devices like ESP32 with LCD screen, or in CGA video card, or on linux without a desktop.
If you need to speed-up the text control, i would recommend to implement own control with virtualization support, ot may be just check similar stuff for wpf or silverlight.
Let me know, please, if you need more information.

**Reply by guilhermeolisi** [reply]
OK,
I got it, so I need another solution, I'll study control virtualization
Thanks for your response
